packages/scaffold-ui/src/partials/w3m-social-login-widget/index.ts

```diff
@@ -217,7 +226,9 @@ export class W3mSocialLoginWidget extends LitElement {
       }
     }
 
-    if (socialProvider) {
+    if (shouldRedirectToUsageExceededView) {
+      RouterController.push('UsageExceeded')
+    } else if (socialProvider) {
```

Comment:
why are we doing this here (and in many other places) instead of handling it globally? Do we need to redirect from the connection view? 
else it should be enough to have a listener in w3m-modal (which I think you already have)